File: app/src/main/java/com/example/android/evince/MainContract.java
package com.example.android.evince;

import com.example.android.evince.pojo.Matrix;

import java.util.List;

public interface MainContract {

     interface MainView {
         void setRows(int mRows, boolean setViewValue);

         void setColumns(int mColumns, boolean setViewValue);

         void setRandomNumber(int randomNumber, boolean setViewValue);

         void setRandomColor(int randomColor, boolean setViewValue);

         void setDefaultData(int rows, int columns, List<Matrix> matrixList, List<Integer> randomList, int positionOfLastStoredRandomNumberIfAny);

         void setRecyclerView(int rows, int columns, List<Matrix> mList);

         void onReInitRandomList();

         void highLightRandomMatch();
    }

     interface Presenter {
         void handleViews();

         void setRandomNumber(int randomNumber);

         void setRandomColor(int randomColor);

         void saveRandomNumber(int randomNumber);

         void saveRandomColor(int randomColor);

         int getRows();

         int getColumns();

         int getRandomNumber();

         int getRandomColor();

         List<Integer> getRandomList();

         void setPositionOfLastStoredRandomNumberIfAny(int position);

         int getPositionOfLastStoredRandomNumberIfAny();

         void onClickApply(int rows, int columns);

         void onClickRandom();

         void increasePosition();
    }

     interface DataInteractor {

         void setRows(int rows);

         void setColumns(int columns);

         void setRandomNumber(int randomNumber);

         void setRandomColor(int randomColor);

         void saveRows(int rows);

         void saveColumns(int columns);

         void saveRandomNumber(int randomNumber);

         void saveRandomColor(int randomColor);

         int getRows();

         int getColumns();

         int getRandomNumber();

         int getRandomColor();

         void saveMatrixList(List<Matrix> matrixList);

         void setRandomList(List<Integer> randomList);

         List<Integer> getRandomList();

         void setPositionOfLastStoredRandomNumberIfAny(int position);

         int getPositionOfLastStoredRandomNumberIfAny();
    }
}

File: app/src/main/java/com/example/android/evince/MainContract.kt
package com.example.android.evince

import com.example.android.evince.pojo.Matrix

interface MainContract {

    interface MainView {
        fun setRows(mRows: Int, setViewValue: Boolean)

        fun setColumns(mColumns: Int, setViewValue: Boolean)

        fun setRandomNumber(randomNumber: Int, setViewValue: Boolean)

        fun setRandomColor(randomColor: Int, setViewValue: Boolean)

        fun setDefaultData(rows: Int, columns: Int, matrixList: MutableList<Matrix>, randomList: MutableList<Int>, positionOfLastStoredRandomNumberIfAny: Int)

        fun setRecyclerView(rows: Int, columns: Int, mList: MutableList<Matrix>)

        fun onReInitRandomList()

        fun highLightRandomMatch()
    }

    interface Presenter {

        val rows: Int

        val columns: Int

        var randomNumber: Int

        var randomColor: Int

        val randomList: List<Int>

        var positionOfLastStoredRandomNumberIfAny: Int
        fun handleViews()

        fun saveRandomNumber(randomNumber: Int)

        fun saveRandomColor(randomColor: Int)

        fun onClickApply(rows: Int, columns: Int)

        fun onClickRandom()

        fun increasePosition()
    }

    interface DataInteractor {

        var rows: Int

        var columns: Int

        var randomNumber: Int

        var randomColor: Int

        var randomList: List<Int>

        var positionOfLastStoredRandomNumberIfAny: Int

        fun saveRows(rows: Int)

        fun saveColumns(columns: Int)

        fun saveRandomNumber(randomNumber: Int)

        fun saveRandomColor(randomColor: Int)

        fun saveMatrixList(matrixList: MutableList<Matrix>)
    }
}
